src/types/typeobject.rs

```diff
@@ -109,12 +109,18 @@ pub trait PyTypeMethods<'py>: crate::sealed::Sealed {
     /// Retrieves the underlying FFI pointer associated with this Python object.
     fn as_type_ptr(&self) -> *mut ffi::PyTypeObject;
 
-    /// Gets the full name, which includes the module, of the `PyType`.
-    fn name(&self) -> PyResult<Cow<'_, str>>;
+    /// Gets the short name of the `PyType`.
+    fn name(&self) -> PyResult<String>;
 
     /// Gets the [qualified name](https://docs.python.org/3/glossary.html#term-qualified-name) of the `PyType`.
     fn qualname(&self) -> PyResult<String>;
 
+    /// Gets the name of the module defining the `PyType`.
+    fn module(&self) -> PyResult<String>;
+
+    /// Gets the [fully qualified name](https://docs.python.org/3/glossary.html#term-qualified-name) of the `PyType`.
```

Comment:
Probably better for this to link to the PEP? Maybe https://peps.python.org/pep-0737/#add-pytype-getfullyqualifiedname-function